src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        File writeToFile = new File(toFileName);
+        File writeFromFile = new File(fromFileName);
+        try {
+            writeToFile.createNewFile();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(writeFromFile))) {
+            try (BufferedWriter writer = new BufferedWriter(new FileWriter(writeToFile))) {
+                String[] sumOfTypes = getSumOpType(reader).split(" ");
+                String[] types = new String[]{"supply", "buy", "result"};
+                writer.write(getDataForFile(sumOfTypes,types));
+                writer.flush();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("Can't find a file " + fromFileName, e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from a file", e);
+        }
```

Comment:
Треба розбити задачу на підзадачі (окремі, логічно цільні операції) і для них створити приватні методи.
```suggestion
    String dataFromFile = readFile(fromFileName);
    String report = createReport(dataFromFile);
    writeToFile(report, toFileName);
```